### PR TITLE
cli: add `-json` and `-t` flag for `alloc checks` command 

### PR DESCRIPTION
**Pull request by Juanadelacuesta**

### CODE REVIEW

File: website/content/docs/commands/alloc/checks.mdx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+Name       =  alive
+Group      =  example.cache[0]
+Task       =  (group)
+Service    =  redis-cache
+Status     =  success
+Mode       =  healthiness
+Timestamp  =  2023-03-09T16:10:23+01:00
+Output     =  nomad: tcp ok
+```
+
```
Comment (jrasell) [inline website/content/docs/commands/alloc/checks.mdx:56]: Lets also add an example each for using the JSON and template flag, so new users have the easiest time learning and understanding what is possible.

**Review by jrasell — CHANGES_REQUESTED** [review]
Looking great, thanks!

Along with the inline comments, this PR will need a changelog entry, [this guide](https://github.com/hashicorp/nomad/blob/main/contributing/CHANGELOG.md) details how to add one and  [this PR](https://github.com/hashicorp/nomad/pull/16357/files) can be used as a basis; although this will be an improvement rather than a bug.

You'll also need to sign the CLA which is detailed in the PR comments, before I can merge this in.

File: command/alloc_checks.go [hunk 2]
```diff
@@ -30,6 +30,9 @@ Checks Specific Options:
 
   -verbose
     Show full information.
+
+  -json
+    Output the latest health check status information in a JSON format.
```
[the remark this answers is not in the document]
Comment (Juanadelacuesta) [reply command/alloc_checks.go:35]: Adding it!

File: command/alloc_checks.go [hunk 3]
```diff
@@ -42,6 +45,7 @@ func (c *AllocChecksCommand) AutocompleteFlags() complete.Flags {
 	return mergeAutocompleteFlags(c.Meta.AutocompleteFlags(FlagSetClient),
 		complete.Flags{
 			"-verbose": complete.PredictNothing,
+			"-json":    complete.PredictNothing,
```
[the remark this answers is not in the document]
Comment (Juanadelacuesta) [reply command/alloc_checks.go:51]: also on its way! :)

File: command/alloc_checks_test.go [hunk 4]
```diff
@@ -117,4 +120,15 @@ func TestAllocChecksCommand_Run(t *testing.T) {
...
+	ui.OutputWriter.Reset()
+	// List json
+	cmd = &AllocChecksCommand{Meta: Meta{Ui: ui, flagAddress: url}}
+	must.Zero(t, cmd.Run([]string{"-address=" + url, "-json", allocID}))
+
+	outJson := api.AllocCheckStatuses{}
+	err = json.Unmarshal(ui.OutputWriter.Bytes(), &outJson)
+	must.NoError(t, err)
+
+	ui.OutputWriter.Reset()
```
[the remark this answers is not in the document]
Comment (Juanadelacuesta) [reply command/alloc_checks_test.go:134]: Consider it done :)

File: website/content/docs/commands/alloc/checks.mdx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+```plaintext
+nomad alloc checks [options] <allocation>
+```
+
+Outputs the latest health check status information for services in the allocation
+using the Nomad service discovery provider. This command accepts an allocation
+ID or prefix as the sole argument.
+
+When ACLs are enabled, this command requires a token with the read-job and list-jobs 
+capabilities for the allocation's namespace.
```
[the remark this answers is not in the document]
Comment (Juanadelacuesta) [reply website/content/docs/commands/alloc/checks.mdx]: Ok, let me fix it :)

File: website/content/docs/commands/alloc/checks.mdx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+## General Options
+
+@include 'general_options.mdx'
+
+## Exec Options
+
+- `-verbose`: Display verbose output.
+
+- `-json`: Output the allocation in its JSON format.
```
[the remark this answers is not in the document]
Comment (Juanadelacuesta) [reply website/content/docs/commands/alloc/checks.mdx:35]: Will do!

File: website/content/docs/commands/alloc/checks.mdx (hunk 1 above)
Comment (Juanadelacuesta) [reply website/content/docs/commands/alloc/checks.mdx:56]: Good idea

File: modd.conf [hunk 7]
```diff
@@ -0,0 +1,4 @@
+command/*.go !**/*_test.go  {
+    prep: make dev
+    daemon +sigterm: bin/nomad agent -dev -bind 0.0.0.0 -log-level INFO
+}
```
Comment (jrasell) [inline modd.conf:4]: Do we need this file committed, or was this for testing?

File: website/content/docs/commands/alloc/checks.mdx (hunk 6 above)
[the remark this answers is not in the document]
Comment (jrasell) [reply website/content/docs/commands/alloc/checks.mdx:35]: I think this is the only outstanding item.

File: modd.conf (hunk 7 above)
Comment (Juanadelacuesta) [reply modd.conf:4]: Ups, my bad! I use it for local development. Removing it!

**Review by jrasell — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**